### PR TITLE
[CAZ-2043] Payment status screen

### PR DESCRIPTION
**Pull request by musztardem**
## Motivation and Context

[CAZ-2043 JIRA Card](https://eaflood.atlassian.net/browse/CAZ-2043)

## Description

Implemented flow:
* After confirming payment data, a request is sent to Payments API.
* Response coming from Payments API returns URL on which user should be redirected.
* Redirection is going to an action that performs another call to Payments API in order to get payments data.
    * When payment is successfully processed - redirection is made to success result page.
    * Else -> redirection is made to failure page.

## How Has This Been Tested?

* RSpec + cucumber test suites
* Manually

### Success page:
Screenshot 2020-03-12 at 15 48 19

----

### Failure page:
Screenshot 2020-03-12 at 15 49 19

## Checklist:

- [x] It contains only changes required by issue (does not contain other PR)
- [x] Includes link to an issue (if apply)
- [x] I have added tests to cover my changes.

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -407,4 +407,4 @@ RUBY VERSION
    ruby 2.6.5p114
 
 BUNDLED WITH
-   2.0.2
+   2.1.2
```
Comment (likeuwill) [inline Gemfile.lock]: Please revert bundle version, we can't use more than 2.0.2 on production

File: app/views/payments/failure.html.haml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+- content_for(:title, service_name)
```
Comment (likeuwill) [inline app/views/payments/failure.html.haml]:
```
- title_and_header = 'Payment unsuccessful''
- content_for(:title, title_and_header)
...
      %h1.govuk-heading-l
        = title_and_header
```

File: app/views/payments/success.html.haml [hunk 3]
```diff
@@ -0,0 +1,55 @@
+- content_for(:title, service_name)
```
Comment (likeuwill) [inline app/views/payments/success.html.haml]: same

File: spec/api_wrappers/payments_api/payment_status_spec.rb [hunk 4]
```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'PaymentsApi.payment_status' do
+  subject(:call) { PaymentsApi.payment_status(payment_id: id, caz_name: 'leeds') }
+
+  let(:id) { SecureRandom.uuid }
+  let(:status) { 'success' }
+  let(:email) { 'test@example.com' }
```
Comment (likeuwill) [inline spec/api_wrappers/payments_api/payment_status_spec.rb]: You use these values in tests only once, just move it into context

**Comment by cruddasj** [comment]
Tagging myself in via commentary to know when this gets merged down. Particularly keen to test the payment email integration which this page invokes as part of the call to the payments API sooner rather than later :)

**Pull request merged**